### ISSUE TITLE
Missing WWW-Authenticate on missing Authorisation with a BearerToken ResourceProtector

### ISSUE THREAD

**Issue by ldng** [lepture/authlib#296]
When protecting a resource as follow :

```python
require_oauth = ResourceProtector()
require_oauth.register_token_validator(BearerTokenValidator(OAuth2Token, realm='https://AUTH_SERVICE_URL/token'))

@require_oauth()
def home(request):
    return HttpResponse('Authenticated')
```

I expect a request *without* an "Authorisation" header to return a 401 response with a "WWW-Authenticate" header per https://tools.ietf.org/html/rfc6750#section-3

Currently, the header is not present in the response.

Authlib version 0.15.2

**Comment by ldng** [comment]
So as far as I understand, the `authlib.oauth2.rfc6749.resource_protector.ResourceProtector` has a `validate_request` method that is called in `authlib.integrations.django_oauth2.ResourceProtector` in the `acquire_token` method.

The `validate_request` method raises a authlib.oauth2.rfc6749.errors.MissingAuthorizationError which does not set the WWW-Authenticate header

Yet the `authlib.oauth2.rfc6750.validator.BearerTokenValidator` has a `request_invalid` to be implemented by subclasses.

Shouldn't this be used instead so that different kind of token could validate the request themselves and report errors on their own terms ?

On the other hand, if I send an invalid token, `authlib.rfc6750.errors.InvalidTokenError` does set the WWW-Authenticate header in its `get_headers` method.

**Comment by ldng** [comment]
Could you at least explain why you consider it not being a bug ? The RFC is pretty clear. Not a book, a oneliner would be fine.

**Comment by lepture** [comment]
@ldng it was fixed by https://github.com/lepture/authlib/commit/62d942e62eefc76397498d0e880bb4198ec9ff22

Check the message above your comment:

> lepture closed this in 62d942e

**Comment by ldng** [comment]
Wow, I missed the hash, apologies ! Many thanks, I need the fix.
